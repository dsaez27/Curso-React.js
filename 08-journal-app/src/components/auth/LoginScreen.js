import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { useForm } from './../../hooks/useForm';
import {
    startGoogleLogin,
    startLoginEmailPassword,
} from '../../redux/actions/auth';

export const LoginScreen = () => {
    const dispatch = useDispatch();
    const { loading } = useSelector((state) => state.ui);

    const [formValues, handleInputChange] = useForm({
        email: 'Daniel',
        password: '123456',
    });

    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        isFormValid && dispatch(startLoginEmailPassword(email, password));
    };

    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin());
    };

    const isFormValid = () => {
        if (email.trim().length === 0) {
            return false;
        } else if (password.trim().length === 0) {
            return false;
        }

        return true;
    };

    return (
        <>
            <h3 className='auth__title'>Login</h3>

            <form onSubmit={handleLogin}>
                <input
                    className='auth__input'
                    type='text'
                    placeholder='Email'
                    name='email'
                    autoComplete='off'
                    value={email}
                    onChange={handleInputChange}
                />
                <input
                    className='auth__input'
                    type='password'
                    placeholder='Password'
                    name='password'
                    autoComplete='off'
                    value={password}
                    onChange={handleInputChange}
                />
                <button
                    className='btn btn-primary btn-block'
                    type='submit'
                    disabled={loading}
                >
                    Login
                </button>
                <div className='auth__social-networks'>
                    <p>Login with social networks</p>
                    <div className='google-btn' onClick={handleGoogleLogin}>
                        <div className='google-icon-wrapper'>
                            <img
                                className='google-icon'
                                src='https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg'
                                alt='google button'
                            />
                        </div>
                        <p className='btn-text'>
                            <b>Sign in with google</b>
                        </p>
                    </div>
                </div>
                <Link to='auth/register' className='link'>
                    Create new account
                </Link>
            </form>
        </>
    );
};
