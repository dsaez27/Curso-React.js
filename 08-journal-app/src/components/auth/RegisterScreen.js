import React from 'react';
import { Link } from 'react-router-dom';

export const RegisterScreen = () => {
    return (
        <>
            <h3 className='auth__title'>Register</h3>

            <form>
                <input
                    className='auth__input mb-1'
                    type='text'
                    placeholder='Name'
                    name='name'
                    autoComplete='off'
                />
                <input
                    className='auth__input mb-1'
                    type='text'
                    placeholder='Email'
                    name='email'
                    autoComplete='off'
                />
                <input
                    className='auth__input mb-1'
                    type='password'
                    placeholder='Password'
                    name='password'
                />
                <input
                    className='auth__input mb-1'
                    type='password'
                    placeholder='Confirm Password'
                    name='password2'
                />
                <button
                    className='btn btn-primary btn-block mb-5'
                    type='submit'
                >
                    Register
                </button>

                <Link to='auth/login' className='link'>
                    Already registered?
                </Link>
            </form>
        </>
    );
};
