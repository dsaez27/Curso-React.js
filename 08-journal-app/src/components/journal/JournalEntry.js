import React from 'react';

export const JournalEntry = () => {
    return (
        <div className='journal__entry pointer'>
            <div
                className='journal__entry-picture'
                style={{
                    backgroundSize: 'cover',
                    backgroundImage:
                        'url(https://i.blogs.es/2e7900/img19_1920x1200/1366_2000.jpg)',
                }}
            ></div>
            <div className='journal__entry-body'>
                <p className='journal__entry-title'>
                    Un nuevo d√≠a
                </p>
                <p className='journal__entry-content'>
                    Lorem Commodo tempor adipisicing proident enim proident labore dolor dolore nostrud ullamco.
                </p>
            </div>
            <div className='journal__entry-date-box'>
                <span>Monday</span>
                <h4>28</h4>
            </div>
        </div>
    );
};
