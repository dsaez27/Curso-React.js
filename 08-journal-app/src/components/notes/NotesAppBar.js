import React from 'react'

export const NotesAppBar = () => {
    return (
        <div className="notes__appbar">
            <span>28 August 2021</span>
            <div>
                <button className='btn'>
                    Pictures
                </button>
            </div>
        </div>
    )
}
