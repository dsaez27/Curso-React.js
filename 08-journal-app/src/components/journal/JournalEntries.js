import React from 'react';
import { JournalEntry } from './JournalEntry';
import { useSelector } from 'react-redux';

export const JournalEntries = () => {
    const { notes } = useSelector((state) => state.notes);

    return (
        <div className='journal__entries'>
            {notes.map((note) => {
                return (
                    <JournalEntry
                        key={note.id}
                        id={note.id}
                        date={note.date}
                        body={note.body}
                        title={note.title}
                        url={note.url}
                    />
                );
            })}
        </div>
    );
};
