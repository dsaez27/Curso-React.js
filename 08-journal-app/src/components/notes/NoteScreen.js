import React from 'react';
import { NotesAppBar } from './NotesAppBar';

export const NoteScreen = () => {
    return (
        <div className='notes__main-content'>
            <NotesAppBar />
            <div className='notes__content'>
                <input
                    className='notes__title-input'
                    type='text'
                    placeholder='Some awesome title'
                    autoComplete='off'
                />
                <textarea
                    placeholder='What happened today?'
                    className='notes__textarea'
                ></textarea>
                <div className='notes__image'>
                    <img src='https://i.blogs.es/2e7900/img19_1920x1200/1366_2000.jpg' alt='img'/>
                </div>
            </div>
        </div>
    );
};
