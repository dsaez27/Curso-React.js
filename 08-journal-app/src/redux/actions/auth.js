import Swal from 'sweetalert2';
import { types } from '../types';
import { finishLoading, setError, startLoading } from './ui';
import { noteLogout } from './notes';
import { firebase, googleAuthProvider } from '../../firebase/firebaseConfig';

export const startLoginEmailPassword = (email, password) => {
    return (dispatch) => {
        dispatch(startLoading());
        try {
            return firebase
                .auth()
                .signInWithEmailAndPassword(email, password)
                .then(({ user }) => {
                    dispatch(login(user.uid, user.displayName));
                    dispatch(finishLoading());
                })
                .catch((e) => {
                    dispatch(finishLoading());
                    Swal.fire({
                        title: 'Error',
                        text: e.message,
                        icon: 'error',
                    });
                });
        } catch (e) {
            console.log(e);
        }
    };
};

export const startRegisterEmailPasswordName = (email, password, name) => {
    return async (dispatch) => {
        try {
            await firebase
                .auth()
                .createUserWithEmailAndPassword(email, password)
                .then(async ({ user }) => {
                    await user.updateProfile({
                        displayName: name,
                    });

                    dispatch(login(user.uid, user.displayName));
                })
                .catch(({ message }) => {
                    Swal.fire({ title: 'Error', text: message, icon: 'error' });
                    dispatch(setError(message));
                });
        } catch (e) {
            console.log(e);
        }
    };
};

export const startGoogleLogin = () => {
    return async (dispatch) => {
       
            await firebase
                .auth()
                .signInWithPopup(googleAuthProvider)
                .then(({ user }) => {
                    dispatch(login(user.uid, user.displayName));
                })
                .catch((error) => {
                    console.log(error);
                });
         
    };
};

export const login = (uid, displayName) => {
    return {
        type: types.login,
        payload: {
            uid,
            displayName,
        },
    };
};

export const startLogout = () => {
    return (dispatch) => {
        firebase.auth().signOut();

        dispatch(logout());
        dispatch(noteLogout());
    };
};

export const logout = () => ({
    type: types.logout,
});
