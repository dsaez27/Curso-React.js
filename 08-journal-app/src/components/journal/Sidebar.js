import React from 'react';
import { JournalEntries } from './JournalEntries';
import { useDispatch, useSelector } from 'react-redux';
import { startLogout } from '../../redux/actions/auth';
import { startNewNote } from '../../redux/actions/notes';

export const Sidebar = () => {
    const dispatch = useDispatch();
    const {name} = useSelector((state) => state.auth);

    const handleLogout = () => {
        dispatch(startLogout());
    };

    const handleAddNew = () => {
        dispatch(startNewNote());
    }

    return (
        <aside className='journal__sidebar'>
            <div className='journal__sidebar-navbar'>
                <h3 className='mt-5'>
                    <i className='far fa-moon'></i>
                    <span className='journal__sidebar-span'>{name}</span>
                </h3>
                <button className='btn mt-5' onClick={handleLogout}>
                    Logout
                </button>
            </div>
            <div className='journal__new-entry' onClick={handleAddNew}>
                <i className='fas fa-calendar-plus fa-5x'></i>
                <span className='mt-5'>New Entry</span>
            </div>
            <JournalEntries />
        </aside>
    );
};
