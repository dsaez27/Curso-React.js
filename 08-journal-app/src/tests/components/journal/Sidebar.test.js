import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import '../../../setupTest';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Sidebar } from './../../../components/journal/Sidebar';
import { startLogout } from '../../../redux/actions/auth';
import { startNewNote } from '../../../redux/actions/notes';

jest.mock('../../../redux/actions/auth', () => ({
    startLogout: jest.fn(),
}));

jest.mock('../../../redux/actions/notes', () => ({
    startNewNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: '123',
        name: 'test',
    },
    ui: {
        loading: false,
        msgError: null,
    },
    notes: {
        active: {
            id: '1',
        },
        notes: [],
    },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <Sidebar />
    </Provider>
);

describe('Pruebas en el Sidebar', () => {
    beforeEach(() => {
        store = mockStore(initState);
        jest.clearAllMocks();
    });

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de llamar el logout', () => {
        wrapper.find('button').prop('onClick')();
        expect(startLogout).toHaveBeenCalled();
    });

    test('Debe de llamar el newNotes', () => {
        wrapper.find('.journal__new-entry').prop('onClick')();
        expect(startNewNote).toHaveBeenCalled();
    });
});
