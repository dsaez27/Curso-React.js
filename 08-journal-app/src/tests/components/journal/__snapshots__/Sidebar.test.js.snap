// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Sidebar Debe de mostrarse correctamente 1`] = `
<aside
  className="journal__sidebar"
>
  <div
    className="journal__sidebar-navbar"
  >
    <h3
      className="mt-5"
    >
      <i
        className="far fa-moon"
      />
      <span
        className="journal__sidebar-span"
      >
        test
      </span>
    </h3>
    <button
      className="btn mt-5"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal__new-entry"
    onClick={[Function]}
  >
    <i
      className="fas fa-calendar-plus fa-5x"
    />
    <span
      className="mt-5"
    >
      New Entry
    </span>
  </div>
  <div
    className="journal__entries"
  />
</aside>
`;
